### DISCUSSION TITLE
PF4

### DISCUSSION THREAD

**Discussion by Wouterdek** [ImageMagick/ImageMagick#2882] [Development]
I see ImageMagick already has support for PFM, but it is currently limited to monochrome or RGB. To add RGBA support, an extension exists called PF4 which adds an extra alpha channel.
You can find more details here: https://www.cse.cuhk.edu.hk/~ttwong/data/hdrfire/hdrfire.html

The only change compared to PFM is the magic keyword in the header, and the extra float per pixel, so adding support should be fairly easy. I've already created [a PR](https://github.com/Tom94/tev/pull/110/files) for an image viewer, which adds support in less than 20 lines.

Thank you for your work on ImageMagick!

**Comment by dlemstra** [comment]
Support for reading those files will be available in the next release of ImageMagick 7.

**Reply by Wouterdek** [reply]
Great, thanks!
